src/actions/plugins/leap/leap.ts

```diff
@@ -0,0 +1,238 @@
+import { Mode } from '../../../mode/mode';
+import { Match } from './match';
+import { Marker, createMarker, generateMarkerNames } from './Marker';
+import { VimState } from '../../../state/vimState';
+import { LeapAction } from './LeapAction';
+import { Position } from 'vscode';
+import { configuration } from '../../../configuration/configuration';
+
+export enum LeapSearchDirection {
+  Forward = -1,
+  Backward = 1,
+}
+
+export class Leap {
+  markers: Marker[] = [];
+  vimState: VimState;
+  previousMode!: Mode;
+  isRepeatLastSearch: boolean = false;
+  direction?: LeapSearchDirection;
+  leapAction?: LeapAction;
+  firstSearchString?: string;
+  searchMode: string = '';
+
+  constructor(vimState: VimState) {
+    this.vimState = vimState;
+  }
+
+  public createMarkers(matches: Match[]) {
+    this.markers = matches.map((match) => {
+      return createMarker(match, this.vimState.editor);
+    });
+
+    this.setMarkersName();
+
+    return this.markers;
+  }
+
+  private setMarkersName() {
+    const map: { [key: string]: Marker[] } = {};
```

Comment:
Let's use a `Map` here, rather than a simple object - I think it's cleaner than relying on `Object.keys` for instance